src/Aspire.Hosting.Azure.CognitiveServices/AzureOpenAIExtensions.cs

```diff
@@ -65,6 +68,17 @@ public static IResourceBuilder<AzureOpenAIResource> AddAzureOpenAI(this IDistrib
                 cdkDeployment.AssignProperty(x => x.Sku.Name, $"'{deployment.SkuName}'");
                 cdkDeployment.AssignProperty(x => x.Sku.Capacity, $"{deployment.SkuCapacity}");
                 cdkDeployments.Add(cdkDeployment);
+
+                // Subsequent deployments need an explicit dependency on the previous one
+                // to ensure they are not created in parallel. This is equivalent to @batchSize(1)
+                // which can't be defined with the CDK
```

Comment:
> This is equivalent to @batchSize(1)
                // which can't be defined with the CDK

When will this be available? That seems like the better long-term fix.

Reply:
We are waiting for this to merge and a build to be produced:

https://github.com/Azure/azure-sdk-for-net/pull/43285/files

@JoshLove-msft can talk to timelines.

Reply:
https://www.nuget.org/packages/Azure.Provisioning/0.2.0-beta.2

Reply:
Thanks, reacting ASAP

Reply:
> When will this be available? That seems like the better long-term fix.

I think the question was not about the `AddDependency` being made public, but the support for `@batchSize(1)`.